File: phoenix-privilege/phoenix-privilege-api/src/main/java/com/phoenix/privilege/entity/PrivilegeModule.java
package com.phoenix.privilege.entity;

import com.mybatisflex.annotation.Table;
import com.phoenix.privilege.base.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigInteger;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Table("tbl_privilege_module")
public class PrivilegeModule extends BaseEntity {

	/** 名称 */
	private String name;

	/** 链接 */
	private String url;

	/** 标识 */
	private String sn;

	/** 存放该模块有哪些权限值可选 */
	private Integer state;

	private String component;

	/** 系统id */
	private Long systemId;

	private Integer status;

	/** 图片路径 */
	private String image;

	/** 模块的排序号 */
	private Integer orderNo;

	private Integer isShow;

	/** 父模块id */
	private String pid;

	/** 权限id */
	private Integer categoryId;

	/** 0目录，1菜单 */
	private String type;

}

File: web-frontend/apps/admin-ui/src/api/core/privilege-module.ts
import { requestClient } from '#/api/request';
import type { PageResult } from '#/api/core/privilege-types';
import type { AclPvalueItem } from '#/api/core/privilege-role';

export interface PrivilegeModule {
  id?: string;
  pid?: string;
  name?: string;
  url?: string;
  sn?: string;
  component?: string;
  image?: string;
  orderNo?: number;
  type?: string;
  status?: number;
  systemId?: number;
  isShow?: number;
  state?: string;
  pvalues?: AclPvalueItem[];
  children?: PrivilegeModule[];
}

export interface ModuleTreeItem {
  id?: string;
  name?: string;
  sn?: string;
  children?: ModuleTreeItem[];
}

export async function getModulePageApi(
  page: number,
  size: number,
  params?: Record<string, any>,
) {
  return requestClient.get<PageResult<PrivilegeModule>>(
    '/api/privilege/module/page',
    {
      params: { page, size, ...params },
      responseReturn: 'body',
    },
  );
}

export async function getModuleTreeApi(systemId?: number) {
  return requestClient.get<PrivilegeModule[]>('/api/privilege/module/tree', {
    params: systemId === undefined ? {} : { systemId },
    responseReturn: 'body',
  });
}

export async function getModuleApi(id: string) {
  return requestClient.get<PrivilegeModule>(`/api/privilege/module/${id}`, {
    responseReturn: 'body',
  });
}

export async function getModulesBySystemApi(systemId: number) {
  return requestClient.get<PrivilegeModule[]>(
    `/api/privilege/module/system/${systemId}`,
    { responseReturn: 'body' },
  );
}

export async function getModulesByPidApi(pid: string) {
  return requestClient.get<PrivilegeModule[]>(
    `/api/privilege/module/pid/${pid}`,
    { responseReturn: 'body' },
  );
}

export async function createModuleApi(data: Record<string, any>) {
  return requestClient.post('/api/privilege/module', data, {
    responseReturn: 'body',
  });
}

export async function updateModuleApi(data: Record<string, any>) {
  return requestClient.put('/api/privilege/module', data, {
    responseReturn: 'body',
  });
}

export async function deleteModuleApi(id: string) {
  return requestClient.delete(`/api/privilege/module/${id}`, {
    responseReturn: 'body',
  });
}
